### PR TITLE
feat: add support for xsl parameters

### LINKED ISSUE #33: XSL parameters
Some XSL will require parameters set by the user.
Cf eg https://ecospheres.gitbook.io/recommandations-iso-dcat/adaptation-des-metadonnees-iso-19139-pour-faciliter-la-transformation-en-dcat/rendre-les-distributions-identifiables would require some sort of pattern matching dependent on the targeted catalog.

### PR DESCRIPTION
**Pull request by abulte**
Fix #33 

Add support for xsl parameters. Only strings are supported for now. String handling is quite messy, I found I needed to use `etree.XSLT.strparam()` for serialization and `str.strip("'")` 🙈 for deserialization.

Bonus:
- do not pass around the full path to the transformation in the web app, we don't want to leak that
- add the transformation name to the `TransformBatch`, can probably become handy
- correct type for xml content in Batches (bytes)
- add a new XML fixture with `gmd:LanguageCode` and modify the XSLT accordingly

Capture d’écran 2024-09-20 à 18 01 52

### CODE REVIEW

File: ecospheres_migrator/migrator.py [hunk 1]
```diff
@@ -38,6 +46,28 @@ class Transformation:
     def name(self) -> str:
         return self.path.stem
 
+    @cached_property
+    def params(self) -> list[TransformationParam]:
+        xslt = etree.parse(self.path, parser=None)
+        root = xslt.getroot()
+        params = []
+        ns = {"xsl": "http://www.w3.org/1999/XSL/Transform"}
+        for param in root.xpath("//xsl:param", namespaces=ns):
```
Comment (streino) [inline ecospheres_migrator/migrator.py:55]:
```suggestion
        for param in root.xpath("//xsl:param", namespaces=root.nsmap):
```
I didn't test specifically here but `root.nsmap` contains all namespaces declared in the parased file, so this should be enough.

File: tests/fixtures/datara_c375f298-bbc4-455b-98ad-f65a20b2e0a9.xml
Comment (streino) [inline tests/fixtures/datara_c375f298-bbc4-455b-98ad-f65a20b2e0a9.xml:1]: I've started using `<catalog>--<id>.xml` in [xslt tests](https://github.com/ecolabdata/ecospheres-xslt/tree/main/fixtures).
Let's pick one and change the other. I'm not fond of `--` but my guess is any single char is likely to appear in an id. WDYT?

File: tests/test_transform.py [hunk 2]
```diff
@@ -76,3 +94,14 @@ def patched_get_md_type(self, md):
         assert len(results.successes()) == len(selection)
     elif md_type[1] == "skipped":
         assert len(results.skipped()) == len(selection)
+
+
+def test_load_transformation_params():
```
Comment (streino) [inline tests/test_transform.py:99]: Should also exercise `required=True` somewhere.

File: tests/fixtures/datara_c375f298-bbc4-455b-98ad-f65a20b2e0a9.xml
Comment (abulte) [reply tests/fixtures/datara_c375f298-bbc4-455b-98ad-f65a20b2e0a9.xml:1]: Ok for `--`.

**Review by streino — APPROVED** [review]

**Pull request merged**